app/controllers/api/plans_base_controller.rb

```diff
@@ -112,16 +113,14 @@ def plans_index_path
 
   protected
 
-  def generic_masterize_plan(issuer, assoc)
-    masterize_plan do
-      if @plan.nil? || issuer.send(assoc) == @plan
-        issuer.send("#{assoc}=", nil)
-      else
-        issuer.send("#{assoc}=", @plan)
-      end
+  def assign_plan!(issuer, assoc)
+    plan = (!@plan || issuer.send(assoc) == @plan) ? nil : @plan
```

Comment:
Basically if it is the same plan passed then you would unset it? That looks strange to me.

Reply:
@guicassolato

Reply:
I haven's changed the logic of it.